data-prepper-plugins/s3-sink/src/main/java/org/opensearch/dataprepper/plugins/sink/s3/KeyGenerator.java

```diff
@@ -35,7 +35,8 @@ public KeyGenerator(final S3SinkConfig s3SinkConfig,
      */
     public String generateKeyForEvent(final Event event) {
         final String pathPrefix = s3BucketSelector != null ? s3BucketSelector.getPathPrefix() : ObjectKey.buildingPathPrefix(s3SinkConfig, event, expressionEvaluator);
```

Comment:
I think the NPE comes from the call to buildingPathPrefix here, so this change doesn't prevent it